bin/check-log.rb

```diff
@@ -208,7 +208,7 @@ def search_log
         line = get_log_entry(line)
       end
 
-      line = line.encode('UTF-8', invalid: :replace, replace: '')
+      line = line.encode('UTF-16', invalid: :replace, replace: '').encode('UTF-8', invalid: :replace, replace: '')
```

Comment:
Let's make this an option, while UTF-16 does have some advantages it does also have some disadvantages as outlined here (quick summary): https://stackoverflow.com/questions/4655250/difference-between-utf-8-and-utf-16 to avoid breaking change we could have it default to utf8 and then override where neccessry.